File: Old JAVA/Enumerise/FoodKind.java
package Enumerise;

import java.util.ArrayList;

public enum FoodKind {
	ALL("All", 100),
	
	//Pas de valeur 0, 1/
	
	COMPLET("Complet", 2),
	COMPLEMENTAIRE("Complementary",3),
	MEN("MENAGE",4),
	BARF("BARF",5)
	
	
		;
		


	private String name = "";
	private int coef = 0;

	//Constructeur
	FoodKind(String name, int coef){
	this.name = name;
	this.coef=coef;
	}



	public String nameToString(){
	return name;
	}
	public int getCoef(){
		  return coef;
		}
	
	
public static FoodKind IntToType(int sz){
	FoodKind r=FoodKind.MEN;
	if (sz==0|sz==1) {
		return FoodKind.MEN;
	}
	for (FoodKind ga: FoodKind.values()){
		if (ga.getCoef()==sz){
			return ga;
		}
		
	}
	return r;
}
public static ArrayList<FoodKind> valuesExcept() {
	
	ArrayList<FoodKind>es=new ArrayList<FoodKind>();
	for (FoodKind e:FoodKind.values()) {
		if (e!=FoodKind.ALL) {
			es.add(e);
		}
			
	}
	return es;
}

@Override
public String toString() {
	return name;
}
	
}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/FoodKind.kt
package fr.vetbrain.vetnutri_mp.Enumer

import fr.vetbrain.vetnutri_mp.Data.Labelable
import kotlinx.serialization.Serializable

@Serializable
enum class FoodKind(val coef: Int, override val label: String) : Labelable {
    ALL(100, "all"),
    COMPLET(2, "complete"),
    COMPLEMENTAIRE(3, "complementary"),
    MEN(4, "household"),
    BARF(5, "barf");

    fun nameToString() = label

    companion object {
        fun byCoef(coef: Int): FoodKind = entries.find { it.coef == coef } ?: MEN

        fun valuesExcept(): List<FoodKind> = entries.filter { it != ALL }
    }

    override fun toString() = label
}
